### ISSUE TITLE
[ecs] Escape hatches do not work for FargateService

### ISSUE THREAD

**Issue by justin8** [aws/aws-cdk#10666]
When attempting to override any property of the underlying CfnResource it appears that `.node.defaultChild` is always undefined.

### Reproduction Steps
```
const service = new ecs.FargateService(...)
const cfnService = service.node.defaultChild as ecs.CfnService;
cfnService.addPropertyDeletionOverride("LaunchType");
```

### What did you expect to happen?

To be able to override properties

### What actually happened?

```
$ cdk synth
...
Cannot read property 'addPropertyDeletionOverride' of undefined
```

### Environment

  - **CLI Version      :** 1.66
  - **Framework Version:** 1.66
  - **Node.js Version:** v14.8.0
  - **OS               :** Linux
  - **Language (Version):** TypeScript (3.8.3)

### Other

--- 

This is :bug: Bug Report

**Comment by Ancient87** [comment]
@justin8 
I came across this issue myself today. While it seems it's still not fixed yet I did find a workaround and I wanted to share what I found for the next person who ends up here looking for a solution.

If you dig into the code of CDK you find out what [defaultChild](https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/core/lib/construct-compat.ts#L362) actually does. It will find a child node named either "Resource" or "Default" in the construct. However, unfortunately, the FargateService doesn't have such a resource and thus it fails.

The workaround is to make use of the [children()](https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/core/lib/construct-compat.ts#L379) accessor method which returns an array of all children of the node. In this case there is only a single one (and arguably should be made the default) of type CfnService. This is also the one you want if you want to make changes to the construct's underlying Cfn resource.

I was able to use the following code to add the "FARGATE_SPOT" capacity provider to an _"ApplicationLoadBalancedFargateService"_

```
const serviceConstruct: ecs.FargateService = this.service.service;
   
    const scn: cdk.ConstructNode = serviceConstruct.node;
    const theFirst: ecs.CfnService = scn.children[0] as ecs.CfnService;
    const strategy = [{capacityProvider: "FARGATE_SPOT", weight:1}];
    theFirst.addPropertyOverride("capacityProviderStrategy", strategy);
```

Hope it helps.
